ecoscope/base/trajectory.py

```diff
@@ -0,0 +1,402 @@
+import itertools
+import warnings
+from copy import deepcopy
+
+import geopandas as gpd  # type: ignore[import-untyped]
+import numpy as np
+import pandas as pd
+import shapely
+from pyproj import Geod
+
+from ecoscope.base._dataclasses import (
+    ProximityProfile,
+    TrajSegFilter,
+)
+from ecoscope.base.straightrack import StraightTrackProperties
+from ecoscope.base import EcoDataFrame, Relocations
+
+
+def get_displacement(gdf: gpd.GeoDataFrame):
+    """
+    Get displacement in meters between first and final fixes.
+    """
+    if not gdf["segment_start"].is_monotonic_increasing:
+        gdf = gdf.sort_values("segment_start")
+    start = gdf.geometry.iloc[0].coords[0]
+    end = gdf.geometry.iloc[-1].coords[1]
+    return Geod(ellps="WGS84").inv(start[0], start[1], end[0], end[1])[2]
+
+
+def get_tortuosity(gdf: gpd.GeoDataFrame):
+    """
+    Get tortuosity for dataframe defined as distance traveled divided by displacement between first and final
+    points.
+    """
+
+    return gdf["dist_meters"].sum() / get_displacement(gdf)
+
+
+class Trajectory(EcoDataFrame):
+    """
+    A trajectory represents a time-ordered collection of segments.
+    Currently only straight track segments exist.
+    It is based on an underlying relocs object that is the point representation
+    """
+
+    @classmethod
+    def from_relocations(cls, relocs: Relocations, *args, **kwargs):
+        """
+        Create Trajectory class from Relocation dataframe.
+        Parameters
+        ----------
+        relocs: Relocations
+            A `Relocations` instance.
+        args
+        kwargs
+        Returns
+        -------
+        Trajectory
+        """
+        assert isinstance(relocs, Relocations)
+        assert {"groupby_col", "fixtime", "geometry"}.issubset(relocs.gdf)
+
+        if kwargs.get("copy"):
+            relocs = Relocations(relocs.gdf.copy())
+
+        original_crs = relocs.gdf.crs
+        relocs.gdf.to_crs(4326, inplace=True)
+        relocs.gdf = (
+            relocs.gdf.groupby("groupby_col")[relocs.gdf.columns].apply(cls._create_multitraj).droplevel(level=0)
+        )
+        relocs.gdf.to_crs(original_crs, inplace=True)
+
+        relocs.gdf.sort_values("segment_start", inplace=True)
+        return Trajectory(gdf=relocs.gdf)
```

Comment:
```suggestion
        return cls(gdf=relocs.gdf)
```

Same comment as for `Relocations` classmethod.